Marlin/src/feature/fwretract.cpp

```diff
@@ -130,47 +126,39 @@ void FWRetract::retract(const bool retracting
 
   const float old_feedrate_mm_s = feedrate_mm_s,
               renormalize = RECIPROCAL(planner.e_factor[active_extruder]),
-              base_retract = swapping ? swap_retract_length : retract_length,
-              old_z = current_position[Z_AXIS],
-              old_e = current_position[E_AXIS];
+              feedrate_scale = 100.0 / feedrate_percentage, // Disable feedrate scaling for retract moves
+              base_retract = swapping ? swap_retract_length : retract_length;
 
   // The current position will be the destination for E and Z moves
   set_destination_from_current();
 
   if (retracting) {
     // Retract by moving from a faux E position back to the current E position
-    feedrate_mm_s = retract_feedrate_mm_s;
-    destination[E_AXIS] -= base_retract * renormalize;
+    feedrate_mm_s = retract_feedrate_mm_s * feedrate_scale;
```

Comment:
I don't personally think that the feedrate multiplier should be applied to the Z hop or retract/recover.

Reply:
Ah! But of course!